src/test/java/org/jenkinsci/plugins/workflow/support/pickles/ThrowablePickleTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2018 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.support.pickles;
+
+import hudson.remoting.ProxyException;
+import java.util.logging.Level;
+import org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.Whitelisted;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.test.steps.SemaphoreStep;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.Rule;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.LoggerRule;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+public class ThrowablePickleTest {
```

Comment:
Not sure which change you are referring to. I only know about a change to `groovy-cps`, which is irrelevant here since we are discussing a binary (Java) class, not a user Groovy class processed by `CpsTransformer`.

Anyway, if you comment out the `@Extension` and run the test:

```
Running org.jenkinsci.plugins.workflow.support.pickles.ThrowablePickleTest
=== Starting smokes(org.jenkinsci.plugins.workflow.support.pickles.ThrowablePickleTest)
   0.071 [id=15]	INFO	o.jvnet.hudson.test.WarExploder#explode: Exploding …/.m2/repository/org/jenkins-ci/main/jenkins-war/2.62/jenkins-war-2.62.war into …/workflow-support-plugin/target/jenkins-for-test
   1.618 [id=15]	INFO	o.jvnet.hudson.test.JenkinsRule#createWebServer: Running on http://localhost:…/jenkins/
   2.148 [id=22]	INFO	jenkins.InitReactorRunner$1#onAttained: Started initialization
   2.911 [id=28]	WARNING	hudson.ClassicPluginStrategy#createClassJarFromWebInfClasses: Created …/jsch/WEB-INF/lib/classes.jar; update plugin to a version created with a newer harness
   2.936 [id=23]	INFO	hudson.PluginManager$1$3$1#isDuplicate: Ignoring …/workflow-support.jpi because …/the.jpl is already loaded
   2.952 [id=22]	INFO	jenkins.InitReactorRunner$1#onAttained: Listed all plugins
   5.564 [id=28]	INFO	jenkins.InitReactorRunner$1#onAttained: Prepared all plugins
   5.564 [id=25]	INFO	jenkins.InitReactorRunner$1#onAttained: Started all plugins
   5.566 [id=25]	INFO	jenkins.InitReactorRunner$1#onAttained: Augmented all extensions
   6.505 [id=25]	INFO	jenkins.InitReactorRunner$1#onAttained: Loaded all jobs
   6.760 [id=21]	INFO	jenkins.InitReactorRunner$1#onAttained: Completed initialization
   6.898 [p #1] Started
   6.900 [p #1] Running in Durability level: MAX_SURVIVABILITY
Blocking wait/1
   7.705 [id=15]	INFO	jenkins.model.Jenkins#cleanUp: Stopping Jenkins
   7.712 [id=15]	INFO	jenkins.model.Jenkins$21#onAttained: Started termination
   7.714 [id=56]	WARNING	h.u.ExceptionCatchingThreadFactory#uncaughtException: Thread Computer.threadPoolForRemoting [#2] terminated unexpectedly
java.nio.channels.ClosedSelectorException
	at sun.nio.ch.SelectorImpl.keys(SelectorImpl.java:68)
	at org.jenkinsci.remoting.protocol.IOHub.run(IOHub.java:422)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
   7.756 [id=43]	WARNING	o.j.p.w.cps.CpsThreadGroup#saveProgramIfPossible: program state save failed
an exception which occurred:
	in field org.jenkinsci.plugins.workflow.support.pickles.ThrowablePickleTest$BadException.notSerializable
	in object org.jenkinsci.plugins.workflow.support.pickles.ThrowablePickleTest$BadException@3abb3d85
	in field com.cloudbees.groovy.cps.impl.BlockScopeEnv.locals
	in object com.cloudbees.groovy.cps.impl.BlockScopeEnv@3dae8055
	in field com.cloudbees.groovy.cps.impl.ProxyEnv.parent
	in object com.cloudbees.groovy.cps.impl.TryBlockEnv@42eab056
	in field com.cloudbees.groovy.cps.impl.ProxyEnv.parent
	in object com.cloudbees.groovy.cps.impl.BlockScopeEnv@57b62a9a
	in field com.cloudbees.groovy.cps.impl.CallEnv.caller
	in object com.cloudbees.groovy.cps.impl.FunctionCallEnv@52898d43
	in field com.cloudbees.groovy.cps.Continuable.e
	in object org.jenkinsci.plugins.workflow.cps.SandboxContinuable@1df2ed52
	in field org.jenkinsci.plugins.workflow.cps.CpsThread.program
	in object org.jenkinsci.plugins.workflow.cps.CpsThread@b7f775f
	in field org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.threads
	in object org.jenkinsci.plugins.workflow.cps.CpsThreadGroup@6422d329
	in object org.jenkinsci.plugins.workflow.cps.CpsThreadGroup@6422d329
Caused: java.io.NotSerializableException: java.lang.Object
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:763)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.BlockMarshaller.doWriteObject(BlockMarshaller.java:65)
	at org.jboss.marshalling.river.BlockMarshaller.writeObject(BlockMarshaller.java:56)
	at org.jboss.marshalling.MarshallerObjectOutputStream.writeObjectOverride(MarshallerObjectOutputStream.java:50)
	at org.jboss.marshalling.river.RiverObjectOutputStream.writeObjectOverride(RiverObjectOutputStream.java:179)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:344)
	at java.util.HashMap.internalWriteEntries(HashMap.java:1790)
	at java.util.HashMap.writeObject(HashMap.java:1363)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.marshalling.reflect.SerializableClass.callWriteObject(SerializableClass.java:273)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:976)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:967)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:967)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:967)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:967)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.BlockMarshaller.doWriteObject(BlockMarshaller.java:65)
	at org.jboss.marshalling.river.BlockMarshaller.writeObject(BlockMarshaller.java:56)
	at org.jboss.marshalling.MarshallerObjectOutputStream.writeObjectOverride(MarshallerObjectOutputStream.java:50)
	at org.jboss.marshalling.river.RiverObjectOutputStream.writeObjectOverride(RiverObjectOutputStream.java:179)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:344)
	at java.util.TreeMap.writeObject(TreeMap.java:2438)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.marshalling.reflect.SerializableClass.callWriteObject(SerializableClass.java:273)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:976)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.AbstractObjectOutput.writeObject(AbstractObjectOutput.java:58)
	at org.jboss.marshalling.AbstractMarshaller.writeObject(AbstractMarshaller.java:111)
	at org.jenkinsci.plugins.workflow.support.pickles.serialization.RiverWriter.lambda$writeObject$0(RiverWriter.java:144)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.runInSandbox(GroovySandbox.java:108)
	at org.jenkinsci.plugins.workflow.support.pickles.serialization.RiverWriter.writeObject(RiverWriter.java:143)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:459)
Caused: java.io.IOException: Failed to persist …/junit884564332800418203/jobs/p/builds/1/program.dat
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:473)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:435)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgramIfPossible(CpsThreadGroup.java:423)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:363)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$100(CpsThreadGroup.java:82)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:243)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:231)
	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:64)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:112)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
   7.808 [p #1] [Pipeline] semaphore
   7.918 [id=58]	INFO	o.j.p.workflow.job.WorkflowRun#finish: p #1 completed: FAILURE
   7.947 [p #1] [Pipeline] End of Pipeline
   7.947 [p #1] an exception which occurred:
   7.948 [p #1] 	in field org.jenkinsci.plugins.workflow.support.pickles.ThrowablePickleTest$BadException.notSerializable
   7.949 [p #1] 	in object org.jenkinsci.plugins.workflow.support.pickles.ThrowablePickleTest$BadException@3abb3d85
   7.949 [p #1] 	in field com.cloudbees.groovy.cps.impl.BlockScopeEnv.locals
   7.949 [p #1] 	in object com.cloudbees.groovy.cps.impl.BlockScopeEnv@3dae8055
   7.950 [p #1] 	in field com.cloudbees.groovy.cps.impl.ProxyEnv.parent
   7.950 [p #1] 	in object com.cloudbees.groovy.cps.impl.TryBlockEnv@42eab056
   7.950 [p #1] 	in field com.cloudbees.groovy.cps.impl.ProxyEnv.parent
   7.950 [p #1] 	in object com.cloudbees.groovy.cps.impl.BlockScopeEnv@57b62a9a
   7.951 [p #1] 	in field com.cloudbees.groovy.cps.impl.CallEnv.caller
   7.951 [p #1] 	in object com.cloudbees.groovy.cps.impl.FunctionCallEnv@52898d43
   7.951 [p #1] 	in field com.cloudbees.groovy.cps.Continuable.e
   7.951 [p #1] 	in object org.jenkinsci.plugins.workflow.cps.SandboxContinuable@1df2ed52
   7.952 [p #1] 	in field org.jenkinsci.plugins.workflow.cps.CpsThread.program
   7.952 [p #1] 	in object org.jenkinsci.plugins.workflow.cps.CpsThread@b7f775f
   7.952 [p #1] 	in field org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.threads
   7.952 [p #1] 	in object org.jenkinsci.plugins.workflow.cps.CpsThreadGroup@6422d329
   7.952 [p #1] 	in object org.jenkinsci.plugins.workflow.cps.CpsThreadGroup@6422d329
   7.953 [p #1] Caused: java.io.NotSerializableException: java.lang.Object
   7.953 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:763)
   7.953 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
   7.953 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
   7.953 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
   7.953 [p #1] 	at org.jboss.marshalling.river.BlockMarshaller.doWriteObject(BlockMarshaller.java:65)
   7.954 [p #1] 	at org.jboss.marshalling.river.BlockMarshaller.writeObject(BlockMarshaller.java:56)
   7.954 [p #1] 	at org.jboss.marshalling.MarshallerObjectOutputStream.writeObjectOverride(MarshallerObjectOutputStream.java:50)
   7.954 [p #1] 	at org.jboss.marshalling.river.RiverObjectOutputStream.writeObjectOverride(RiverObjectOutputStream.java:179)
   7.954 [p #1] 	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:344)
   7.954 [p #1] 	at java.util.HashMap.internalWriteEntries(HashMap.java:1790)
   7.954 [p #1] 	at java.util.HashMap.writeObject(HashMap.java:1363)
   7.955 [p #1] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   7.955 [p #1] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   7.955 [p #1] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   7.955 [p #1] 	at java.lang.reflect.Method.invoke(Method.java:498)
   7.963 [p #1] 	at org.jboss.marshalling.reflect.SerializableClass.callWriteObject(SerializableClass.java:273)
   7.963 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:976)
   7.963 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
   7.963 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
   7.963 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
   7.964 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
   7.964 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
   7.965 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
   7.965 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:967)
   7.965 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
   7.965 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
   7.965 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
   7.965 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:967)
   7.965 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
   7.965 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
   7.966 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
   7.966 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:967)
   7.969 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
   7.969 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
   7.969 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
   7.969 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:967)
   7.970 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
   7.970 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
   7.970 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
   7.970 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
   7.970 [p #1] 	at org.jboss.marshalling.river.BlockMarshaller.doWriteObject(BlockMarshaller.java:65)
   7.970 [p #1] 	at org.jboss.marshalling.river.BlockMarshaller.writeObject(BlockMarshaller.java:56)
   7.970 [p #1] 	at org.jboss.marshalling.MarshallerObjectOutputStream.writeObjectOverride(MarshallerObjectOutputStream.java:50)
   7.970 [p #1] 	at org.jboss.marshalling.river.RiverObjectOutputStream.writeObjectOverride(RiverObjectOutputStream.java:179)
   7.970 [p #1] 	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:344)
   7.971 [p #1] 	at java.util.TreeMap.writeObject(TreeMap.java:2438)
   7.971 [p #1] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   7.971 [p #1] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   7.971 [p #1] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   7.971 [p #1] 	at java.lang.reflect.Method.invoke(Method.java:498)
   7.971 [p #1] 	at org.jboss.marshalling.reflect.SerializableClass.callWriteObject(SerializableClass.java:273)
   7.971 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:976)
   7.971 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
   7.971 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
   7.972 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
   7.972 [p #1] 	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
   7.972 [p #1] 	at org.jboss.marshalling.AbstractObjectOutput.writeObject(AbstractObjectOutput.java:58)
   7.972 [p #1] 	at org.jboss.marshalling.AbstractMarshaller.writeObject(AbstractMarshaller.java:111)
   7.972 [p #1] 	at org.jenkinsci.plugins.workflow.support.pickles.serialization.RiverWriter.lambda$writeObject$0(RiverWriter.java:144)
   7.974 [p #1] 	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.runInSandbox(GroovySandbox.java:108)
   7.974 [p #1] 	at org.jenkinsci.plugins.workflow.support.pickles.serialization.RiverWriter.writeObject(RiverWriter.java:143)
   7.974 [p #1] 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:459)
   7.975 [p #1] 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:435)
   7.975 [p #1] 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgramIfPossible(CpsThreadGroup.java:423)
   7.975 [p #1] 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:363)
   7.975 [p #1] 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$100(CpsThreadGroup.java:82)
   7.975 [p #1] 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:243)
   7.975 [p #1] 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:231)
   7.975 [p #1] 	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:64)
   7.975 [p #1] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
   7.975 [p #1] 	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:112)
   7.976 [p #1] 	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
   7.976 [p #1] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
   7.976 [p #1] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
   7.977 [p #1] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
   7.977 [p #1] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
   7.977 [p #1] 	at java.lang.Thread.run(Thread.java:748)
   7.977 [p #1] Finished: FAILURE
   7.985 [id=15]	INFO	jenkins.model.Jenkins$21#onAttained: Completed termination
   7.985 [id=15]	INFO	jenkins.model.Jenkins#_cleanUpDisconnectComputers: Starting node disconnection
   8.183 [id=62]	INFO	h.TcpSlaveAgentListener$ConnectionHandler#run: Accepted Ping connection #1 from /127.0.0.1:…
   8.185 [id=15]	INFO	jenkins.model.Jenkins#_cleanUpShutdownPluginManager: Stopping plugin manager
   8.185 [id=15]	INFO	jenkins.model.Jenkins#_cleanUpPersistQueue: Persisting build queue
   8.189 [id=15]	INFO	jenkins.model.Jenkins#_cleanUpAwaitDisconnects: Waiting for node disconnection completion
WARN: The method class org.apache.commons.logging.impl.SLF4JLogFactory#release() was invoked.
WARN: Please see http://www.slf4j.org/codes.html#release for an explanation.
   8.189 [id=15]	INFO	jenkins.model.Jenkins#cleanUp: Jenkins stopped
=== Starting smokes(org.jenkinsci.plugins.workflow.support.pickles.ThrowablePickleTest)
   0.028 [id=63]	INFO	o.jvnet.hudson.test.JenkinsRule#createWebServer: Running on http://localhost:…/jenkins/
   0.052 [id=69]	INFO	jenkins.InitReactorRunner$1#onAttained: Started initialization
   0.061 [id=71]	INFO	jenkins.InitReactorRunner$1#onAttained: Listed all plugins
   0.335 [id=71]	INFO	jenkins.InitReactorRunner$1#onAttained: Prepared all plugins
   0.336 [id=70]	INFO	jenkins.InitReactorRunner$1#onAttained: Started all plugins
   0.339 [id=70]	INFO	jenkins.InitReactorRunner$1#onAttained: Augmented all extensions
   0.433 [id=72]	INFO	jenkins.InitReactorRunner$1#onAttained: Loaded all jobs
   0.453 [id=73]	INFO	jenkins.InitReactorRunner$1#onAttained: Completed initialization
Unblocking wait/1 as success
   0.517 [id=63]	INFO	jenkins.model.Jenkins#cleanUp: Stopping Jenkins
   0.519 [id=63]	INFO	jenkins.model.Jenkins$21#onAttained: Started termination
   0.521 [id=63]	INFO	jenkins.model.Jenkins$21#onAttained: Completed termination
   0.521 [id=63]	INFO	jenkins.model.Jenkins#_cleanUpDisconnectComputers: Starting node disconnection
   0.550 [id=94]	INFO	h.TcpSlaveAgentListener$ConnectionHandler#run: Accepted Ping connection #2 from /127.0.0.1:…
   0.551 [id=63]	INFO	jenkins.model.Jenkins#_cleanUpShutdownPluginManager: Stopping plugin manager
   0.552 [id=63]	INFO	jenkins.model.Jenkins#_cleanUpPersistQueue: Persisting build queue
   0.552 [id=63]	INFO	jenkins.model.Jenkins#_cleanUpAwaitDisconnects: Waiting for node disconnection completion
   0.553 [id=63]	INFO	jenkins.model.Jenkins#cleanUp: Jenkins stopped
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 10.029 s <<< FAILURE! - in org.jenkinsci.plugins.workflow.support.pickles.ThrowablePickleTest
smokes(org.jenkinsci.plugins.workflow.support.pickles.ThrowablePickleTest)  Time elapsed: 9.816 s  <<< FAILURE!
java.lang.AssertionError: 
unexpected build status; build log was:
------
Started
Running in Durability level: MAX_SURVIVABILITY
[Pipeline] semaphore
[Pipeline] End of Pipeline
an exception which occurred:
	in field org.jenkinsci.plugins.workflow.support.pickles.ThrowablePickleTest$BadException.notSerializable
	in object org.jenkinsci.plugins.workflow.support.pickles.ThrowablePickleTest$BadException@3abb3d85
	in field com.cloudbees.groovy.cps.impl.BlockScopeEnv.locals
	in object com.cloudbees.groovy.cps.impl.BlockScopeEnv@3dae8055
	in field com.cloudbees.groovy.cps.impl.ProxyEnv.parent
	in object com.cloudbees.groovy.cps.impl.TryBlockEnv@42eab056
	in field com.cloudbees.groovy.cps.impl.ProxyEnv.parent
	in object com.cloudbees.groovy.cps.impl.BlockScopeEnv@57b62a9a
	in field com.cloudbees.groovy.cps.impl.CallEnv.caller
	in object com.cloudbees.groovy.cps.impl.FunctionCallEnv@52898d43
	in field com.cloudbees.groovy.cps.Continuable.e
	in object org.jenkinsci.plugins.workflow.cps.SandboxContinuable@1df2ed52
	in field org.jenkinsci.plugins.workflow.cps.CpsThread.program
	in object org.jenkinsci.plugins.workflow.cps.CpsThread@b7f775f
	in field org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.threads
	in object org.jenkinsci.plugins.workflow.cps.CpsThreadGroup@6422d329
	in object org.jenkinsci.plugins.workflow.cps.CpsThreadGroup@6422d329
Caused: java.io.NotSerializableException: java.lang.Object
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:763)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.BlockMarshaller.doWriteObject(BlockMarshaller.java:65)
	at org.jboss.marshalling.river.BlockMarshaller.writeObject(BlockMarshaller.java:56)
	at org.jboss.marshalling.MarshallerObjectOutputStream.writeObjectOverride(MarshallerObjectOutputStream.java:50)
	at org.jboss.marshalling.river.RiverObjectOutputStream.writeObjectOverride(RiverObjectOutputStream.java:179)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:344)
	at java.util.HashMap.internalWriteEntries(HashMap.java:1790)
	at java.util.HashMap.writeObject(HashMap.java:1363)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.marshalling.reflect.SerializableClass.callWriteObject(SerializableClass.java:273)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:976)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:967)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:967)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:967)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:967)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.BlockMarshaller.doWriteObject(BlockMarshaller.java:65)
	at org.jboss.marshalling.river.BlockMarshaller.writeObject(BlockMarshaller.java:56)
	at org.jboss.marshalling.MarshallerObjectOutputStream.writeObjectOverride(MarshallerObjectOutputStream.java:50)
	at org.jboss.marshalling.river.RiverObjectOutputStream.writeObjectOverride(RiverObjectOutputStream.java:179)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:344)
	at java.util.TreeMap.writeObject(TreeMap.java:2438)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.marshalling.reflect.SerializableClass.callWriteObject(SerializableClass.java:273)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:976)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1032)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:988)
	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:854)
	at org.jboss.marshalling.AbstractObjectOutput.writeObject(AbstractObjectOutput.java:58)
	at org.jboss.marshalling.AbstractMarshaller.writeObject(AbstractMarshaller.java:111)
	at org.jenkinsci.plugins.workflow.support.pickles.serialization.RiverWriter.lambda$writeObject$0(RiverWriter.java:144)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.runInSandbox(GroovySandbox.java:108)
	at org.jenkinsci.plugins.workflow.support.pickles.serialization.RiverWriter.writeObject(RiverWriter.java:143)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:459)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:435)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgramIfPossible(CpsThreadGroup.java:423)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:363)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$100(CpsThreadGroup.java:82)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:243)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:231)
	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:64)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:112)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Finished: FAILURE

------

Expected: is <SUCCESS>
     but: was <FAILURE>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.jvnet.hudson.test.JenkinsRule.assertBuildStatus(JenkinsRule.java:1188)
	at org.jvnet.hudson.test.JenkinsRule.assertBuildStatusSuccess(JenkinsRule.java:1218)
	at org.jenkinsci.plugins.workflow.support.pickles.ThrowablePickleTest.lambda$smokes$1(ThrowablePickleTest.java:60)
	at org.jvnet.hudson.test.RestartableJenkinsRule$3.evaluate(RestartableJenkinsRule.java:195)
	at org.jvnet.hudson.test.RestartableJenkinsRule$6.evaluate(RestartableJenkinsRule.java:235)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:548)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

which matches the behavior observed in JIRA, that the build is broken without running any `catch` / `finally` blocks or the like.

Reply:
Good enough for me then.  👍